drivers/hmis/app/models/hmis/hud/processors/geolocation_processor.rb

```diff
@@ -0,0 +1,46 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Hmis::Hud::Processors
+  class GeolocationProcessor < Base
+    def factory_name
+      :clh_location_factory
+    end
+
+    def schema
+      nil
+    end
+
+    def process(field, value)
+      attribute_name = ar_attribute_name(field)
+
+      if attribute_name == 'coordinates'
+        attribute_value = JSON.parse(value, symbolize_names: true)
+        latitude = attribute_value[:latitude]
+        longitude = attribute_value[:longitude]
+        not_collected_reason = attribute_value[:notCollectedReason]
+        raise ArgumentError, 'Geolocation coordinates in unexpected format' unless (latitude && longitude) || not_collected_reason
+
+        return @processor.send(factory_name).destroy if not_collected_reason
+
+        @processor.send(factory_name).assign_attributes(lat: latitude, lon: longitude)
+      else
+        @processor.send(factory_name).assign_attributes(attribute_name => value)
+      end
+    end
+
+    def assign_metadata
+      clh = @processor.send(factory_name, create: false)
+      return if clh&.destroyed?
+
+      clh&.assign_attributes(
+        source: @processor.enrollment_factory,
+        located_on: @processor.enrollment_factory.entry_date, # entry_date holds the submission time of the form
```

Comment:
Hmm that's true for PIT, but won't be true for, say, collecting geolocation on a current living situation. (In that case, it would be `@processor.current_living_situation.information_date` aka `@processor.owner.information_date`.

Maybe we can solve that by having the CLS form explicitly set the located_on using a form item (`mapping: {record_type:GEOLOCATION, field_name: located_on}`), and have the processor only set it to `@processor.enrollment_factory.entry_date` as a default (if `@hud_values['Geolocation.locatedOn']` doesn't exist..?)

Or- we could just cross this bridge when we implement https://github.com/open-path/Green-River/issues/5726. Just wanted to point it out.

Reply:
I'm in favor of kicking the can. I'll add a comment, though

Reply:
Edit: Per discussion here https://github.com/greenriver/hmis-warehouse/pull/4696#discussion_r1757167999, seems like it would be helpful to update this so that the CLH record holds the timestamp of the form submission, for the sake of marker display on the map. What I've proposed here is to stuff that into the `processed_at` field, even though it feels a bit unintuitive to me since I would expect `processed_at` to hold the processing (admin review) timestamp and not the submission timestamp.

I guess another approach we could take is to migrate the located_on col to be a datetime instead of a date. Or, to add a new col located_at which would hold the full timestamp. @gigxz what do you think?

Reply:
I think I'd be inclined to add a new column.  We could coalesce `located_at` with `located_on` for display purposes.

Reply:
I'll add that coalescence on the locations map PR https://github.com/greenriver/hmis-warehouse/pull/4696.